### PR TITLE
Uf 11204

### PR DESCRIPTION
**Pull request by foobar-1337**
## Types of changes

- [x] Bug fix
- [x] New feature
- [x] Removed feature
- [x] Code style update (formatting etc.)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [x] Yes (I have stepped the version number accordingly)
- [ ] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

**Review by AndreSoderlund — APPROVED** [review]

**Review by MickeWallner — APPROVED** [review]

File: src/main/java/se/sundsvall/casedata/api/model/Address.java [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
-import se.sundsvall.casedata.integration.db.model.enums.AddressCategory;
-
 import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.Size;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
-import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
+import se.sundsvall.casedata.integration.db.model.enums.AddressCategory;
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/casedata/api/model/Address.java:9]: On the subject of API-fixes. Many fields in this have `@Size(max = 255)` should probably be reflected in the api with `maxLenght = 255`

Comment (cheezi0747) [reply src/main/java/se/sundsvall/casedata/api/model/Address.java:9]: This is probably valid for other classes too

File: src/main/java/se/sundsvall/casedata/api/model/MessageResponse.java [hunk 2]
```diff
@@ -17,23 +20,17 @@
...
 	@Schema(description = "The message ID", example = "12")
 	private String messageId;
 
-	@Schema(description = "The errand number", example = "PRH-2022-000001")
-	private String errandNumber;
+	@Schema(description = "The errand ID", example = "123")
+	private Long errandId;
 
 	@Schema(description = "The municipality ID", example = "2281")
 	@Size(max = 255)
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/casedata/api/model/MessageResponse.java:36]: Validation in a response class feels redudant or wrong?

File: pom.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 	<parent>
 		<groupId>se.sundsvall.dept44</groupId>
 		<artifactId>dept44-service-parent</artifactId>
-		<version>6.0.3</version>
+		<version>6.0.4</version>
```
Comment (cheezi0747) [inline pom.xml:7]: 6.0.5

**Comment by foobar-1337** [comment]
Acted on review comments

File: src/main/java/se/sundsvall/casedata/integration/parkingpermit/ParkingPermitClient.java [hunk 4]
```diff
@@ -3,16 +3,14 @@
...
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
-
 import se.sundsvall.casedata.integration.parkingpermit.configuration.ParkingPermitConfiguration;
 
-import generated.se.sundsvall.parkingpermit.StartProcessResponse;
-import io.github.resilience4j.retry.annotation.Retry;
-
 @FeignClient(
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/casedata/integration/parkingpermit/ParkingPermitClient.java:14]: Circutbreaker?

File: src/main/resources/db/migration/V3_6__add_errand_id_to_message_and_attachment_tables.sql [hunk 5]
```diff
@@ -0,0 +1,5 @@
+alter table if exists message add column if not exists errand_id bigint;
+create index if not exists idx_messsage_errand_id on message (errand_id);
+
+update attachment a set a.errand_id = (select e.id from errand e where e.errand_number = a.errand_number) where a.errand_id is null;
+update message m set m.errand_id = (select e.id from errand e where e.errand_number = m.errand_number);
```
Comment (cheezi0747) [inline src/main/resources/db/migration/V3_6__add_errand_id_to_message_and_attachment_tables.sql:5]: missing `where m.errand_id is null` ?

File: src/main/resources/db/migration/V3_6__add_errand_id_to_message_and_attachment_tables.sql [hunk 6]
```diff
@@ -0,0 +1,5 @@
+alter table if exists message add column if not exists errand_id bigint;
```
Comment (cheezi0747) [inline src/main/resources/db/migration/V3_6__add_errand_id_to_message_and_attachment_tables.sql:1]: File is named` V3_6__add_errand_id_to_message_and_attachment_tables.sql`but unless I'm misstaken only message is updated?

Comment (foobar-1337) [reply src/main/resources/db/migration/V3_6__add_errand_id_to_message_and_attachment_tables.sql:1]: Yes, but we adds errand_id:s to both table (both structure/data in message and only data in attachment)

**Review by cheezi0747 — APPROVED** [review]
Missing circutbreakers otherwise 👍

**Comment by foobar-1337** [comment]
Added circuit breakers

**Pull request merged**